### DISCUSSION TITLE
How to set to NAs some segments of a channel

### DISCUSSION THREAD

**Discussion by bnicenboim** [bnicenboim/eeguana#182] [Q&A] [answered]
See also https://github.com/bnicenboim/eeguana/issues/167

**Comment by bnicenboim** [comment, marked as answer]
Example:

eeg_fake <- eeg_lst(signal_tbl = tibble(.id = rep(1:10, each =100),
                                        .sample = sample_int(rep(1:100, 10),500),  
                                        P1 = channel_dbl(rnorm(1000)), 
                                        P3 = channel_dbl(rnorm(1000))))

eeg_mutate(eeg_fake, P3 = case_when(.id %in% 1:5 ~ NA, 
                                    TRUE ~ P3))

**Comment by stonekate** [comment]
Thanks for this. The problem I'm having now is that when there are NAs in the signal table, various functions don't work. I believe it's because there's a check somewhere in datatable that is.double == TRUE, and the NAs violate this. So far I've noticed it affects eeg_artif_minmax() and bind(). And probably related functions like eeg_artif_step(). Here's the error message I get:

![image](https://user-images.githubusercontent.com/12735809/164003985-ed769242-60dc-4d9f-8e76-7e0a9500523b.png)

As a reprex using your code above, I get a shorter version of the error (same problem though I think):

eeg_fake <- eeg_lst(signal_tbl = tibble(.id = rep(1:10, each =100),
.sample = sample_int(rep(1:100, 10),500),
P1 = channel_dbl(rnorm(1000)),
P3 = channel_dbl(rnorm(1000))))

eeg_fake <- eeg_mutate(eeg_fake, P3 = case_when(.id %in% 1:5 ~ NA,
TRUE ~ P3))

eeg_artif_minmax(eeg_fake, .threshold = 150, .window = 150, .unit = "ms", na.rm = TRUE)

![image](https://user-images.githubusercontent.com/12735809/164004285-b7b4d99d-fc86-46b7-a0d0-8ba2b4cb0ba6.png)

**Reply by bnicenboim** [reply]
the problem with the minmax function is the `na.rm` argument that doesn't exist. This `eeg_artif_minmax(eeg_fake, .threshold = 150, .window = 150, .unit = "ms")` works. 
And I can also bind two eeg objects with NAs.
`bind(eeg_fake,eeg_fake)` works as well.

**Reply by stonekate** [reply]
Oh sorry I left the `na.rm` in by accident (was just testing). For my non-fake dataset I get the error message, but maybe there's some other problem with my data. I will look into it some more.
